File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/csharp/fenwick-tree.cs
/**
 * C# Implementation: Fenwick Tree (Binary Indexed Tree)
 * Invariant: Prefix Sum O(log N), Update O(log N), Space O(N)
 */
using System;
using System.Collections.Generic;

namespace Seed.DSA.Core
{
    public class FenwickTreeBinaryIndexedTree<T>
    {
        private readonly List<T> _items = new List<T>();

        // TODO: Implement algorithm operations
        public void Add(T item)
        {
            _items.Add(item);
        }

        public int Count => _items.Count;
    }
}

File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/java/fenwick-tree.java
/**
 * Java Implementation: Fenwick Tree (Binary Indexed Tree)
 * Invariant: Prefix Sum O(log N), Update O(log N), Space O(N)
 */
package seed.dsa.core;

import java.util.*;

public class FenwickTreeBinaryIndexedTree<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}
